### PR TITLE
Refactor(server): db schema mapper

### PR DESCRIPTION
**Pull request by abrl91**
Close: #6651

### Summary
📝♻️🐛

This pull request improves the Prisma schema parser service by refactoring the `Mapper` type and its usage, to avoid name conflicts and enhance readability and debugging. It also updates the corresponding type definitions in `types.ts`.

> _`Mapper` type changed_
> _Nested objects with new keys_
> _Autumn leaves falling_

### Walkthrough
*  Rename the `oldName` property of the `MapperItem` type to `originalName`, and update the variable name accordingly ([link](https://github.com/amplication/amplication/pull/6685/files?diff=unified&w=0#diff-101d4943af6514d1fdfefcb8ee7be3d2c6e0f28cb7621a0d41bc39a06d0f7bd3L410-R412), [link](https://github.com/amplication/amplication/pull/6685/files?diff=unified&w=0#diff-101d4943af6514d1fdfefcb8ee7be3d2c6e0f28cb7621a0d41bc39a06d0f7bd3L537-R548))
*  Change the structure of the `fieldNames`, `fieldTypes`, and `idFields` properties of the `Mapper` type to nested objects, using the new model name, the new field name, and the original field type as keys, and add conditional checks to create the nested objects if they do not exist ([link](https://github.com/amplication/amplication/pull/6685/files?diff=unified&w=0#diff-101d4943af6514d1fdfefcb8ee7be3d2c6e0f28cb7621a0d41bc39a06d0f7bd3L489-R503), [link](https://github.com/amplication/amplication/pull/6685/files?diff=unified&w=0#diff-101d4943af6514d1fdfefcb8ee7be3d2c6e0f28cb7621a0d41bc39a06d0f7bd3L543-R575), [link](https://github.com/amplication/amplication/pull/6685/files?diff=unified&w=0#diff-101d4943af6514d1fdfefcb8ee7be3d2c6e0f28cb7621a0d41bc39a06d0f7bd3L774-R817))
*  Update the access and assignment of the `fieldNames`, `fieldTypes`, and `idFields` properties of the `Mapper` type, using the new keys and the constant `ID_FIELD_NAME` ([link](https://github.com/amplication/amplication/pull/6685/files?diff=unified&w=0#diff-101d4943af6514d1fdfefcb8ee7be3d2c6e0f28cb7621a0d41bc39a06d0f7bd3L689-R720), [link](https://github.com/amplication/amplication/pull/6685/files?diff=unified&w=0#diff-101d4943af6514d1fdfefcb8ee7be3d2c6e0f28cb7621a0d41bc39a06d0f7bd3L796-R848))
*  Update the type definitions of the `Mapper` and the `MapperItem` types in `types.ts`, using more descriptive and specific type aliases ([link](https://github.com/amplication/amplication/pull/6685/files?diff=unified&w=0#diff-e63e35639170ef509c305f88cb604991597f84c7c18a46aba7b00423f7dbfee6L23-R39))
*  Add a debug log statement to print the mapper object in `prismaSchemaParser.service.ts` ([link](https://github.com/amplication/amplication/pull/6685/files?diff=unified&w=0#diff-101d4943af6514d1fdfefcb8ee7be3d2c6e0f28cb7621a0d41bc39a06d0f7bd3R137-R138))

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-server/src/core/prismaSchemaParser/prismaSchemaParser.service.ts [hunk 1]
```diff
@@ -534,17 +549,35 @@ export class PrismaSchemaParserService {
...
+              };
+            } else {
+              mapper.fieldTypes[originalModelName] = {
+                [originalFieldName]: {
+                  [field.fieldType]: {
+                    originalName: field.fieldType,
+                    newName,
+                  },
+                },
+              };
```
Comment (overbit) [inline packages/amplication-server/src/core/prismaSchemaParser/prismaSchemaParser.service.ts:579]: What if ` mapper.fieldTypes[originalModelName] ` exists but ` mapper.fieldTypes[originalModelName][originalFieldName]` doesn't? 
The value `mapper.fieldTypes[originalModelName` will get corrupted and a simple unit test will uncover this.

Try
 ```suggestion
            if (field.fieldType === originalName) {
                mapper.fieldTypes = {
                    ...mapper.fieldTypes,
                    [originalModelName]: {
                        ...mapper.fieldTypes[originalModelName],
                        [originalFieldName]: {
                            ...mapper.fieldTypes[originalModelName][originalFieldName],
                            [field.fieldType]: {
                                originalName: field.fieldType,
                                newName,
                            },
                        },
                    },
                };
            }
```

File: packages/amplication-server/src/core/prismaSchemaParser/prismaSchemaParser.service.ts [hunk 2]
```diff
@@ -686,7 +719,8 @@ export class PrismaSchemaParserService {
 
                 for (const arg of rangeIndexArgArr) {
                   if (typeof arg.name === "string") {
-                    const newFieldName = mapper.fieldNames[arg.name]?.newName;
+                    const newFieldName =
+                      mapper.fieldNames[model.name][arg.name]?.newName;
```
Comment (overbit) [inline packages/amplication-server/src/core/prismaSchemaParser/prismaSchemaParser.service.ts:723]: 💯

File: packages/amplication-server/src/core/prismaSchemaParser/prismaSchemaParser.service.ts [hunk 3]
```diff
@@ -771,10 +805,19 @@ export class PrismaSchemaParserService {
...
+              newName: `${model.name}Id`,
+            };
+          } else {
+            mapper.idFields[model.name] = {
+              [field.name]: {
+                originalName: field.name,
+                newName: `${model.name}Id`,
+              },
+            };
+          }
```
Comment (overbit) [inline packages/amplication-server/src/core/prismaSchemaParser/prismaSchemaParser.service.ts:820]:
```suggestion
          mapper.idFields ={
            ...mapper.idFields,
            [model.name]: {
                ...mapper.idFields[model.name],
                [field.name]: {
                    originalName: field.name,
                    newName: `${model.name}Id`,
                },
            },
          }
```

File: packages/amplication-server/src/core/prismaSchemaParser/prismaSchemaParser.service.ts [hunk 4]
```diff
@@ -793,10 +836,19 @@ export class PrismaSchemaParserService {
...
+              newName: ID_FIELD_NAME,
+            };
+          } else {
+            mapper.idFields[model.name] = {
+              [field.name]: {
+                originalName: field.name,
+                newName: ID_FIELD_NAME,
+              },
+            };
+          }
```
Comment (overbit) [inline packages/amplication-server/src/core/prismaSchemaParser/prismaSchemaParser.service.ts:851]:
```suggestion
          mapper.idFields = {
            ...mapper.idFields,
            [model.name]: {
                ...mapper.idFields[model.name],
                [field.name]: {
                    originalName: field.name,
                    newName: ID_FIELD_NAME,
                },
            },
          }
```

File: packages/amplication-server/src/core/prismaSchemaParser/types.ts [hunk 5]
```diff
@@ -20,15 +20,22 @@ export type PrepareOperationIO = {
   actionContext: ActionContext;
 };
 
+type ModelName = string; // original model name
+type FieldName = string; // original field name
+type FieldTypeName = string; // original field type name
```
Comment (overbit) [inline packages/amplication-server/src/core/prismaSchemaParser/types.ts:25]:
```suggestion
/**
 * Original namodel nameel
 */
export type ModelName = string; 
/**
 * Original field name
 */
export type FieldName = string; 
/**
 * Original field type name
 */
export type FieldTypeName = string; 
```

File: packages/amplication-server/src/core/prismaSchemaParser/types.ts [hunk 6]
```diff
@@ -20,15 +20,22 @@ export type PrepareOperationIO = {
   actionContext: ActionContext;
 };
 
+type ModelName = string; // formatted model name
+type FieldName = string; // formatted field name
+type FieldTypeName = string; // formatted field type name
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-server/src/core/prismaSchemaParser/types.ts]: just mentioning the comment was a mistake. It is the original name

**Review by overbit — APPROVED** [review]

**Pull request merged**